### PR TITLE
Symfony 5 support

### PR DESCRIPTION
**Pull request by thesunrise1983**
These changes are working on Symfony 5.0 and Symfony 4.4, I didn't test on lower versions.

### CODE REVIEW

**Comment by dmaicher** [comment]
Seems this change is incompatible with Symfony 3.4:

https://travis-ci.org/nelmio/NelmioSolariumBundle/jobs/649368661?utm_medium=notification&utm_source=github_status

**Comment by Seldaek** [comment]
Maybe it works if you just remove the type hint? I'm not sure if that'll be allowed by PHP.

**Comment by thesunrise1983** [comment]
> Seems this change is incompatible with Symfony 3.4:
> 
> https://travis-ci.org/nelmio/NelmioSolariumBundle/jobs/649368661?utm_medium=notification&utm_source=github_status

Fixed now by removing the type hint. 
Another solution would be to end Symfony 3.4 support by creating a new version.

**Comment by thesunrise1983** [comment]
@dmaicher All tests passed successfully.

File: composer.json [hunk 1]
```diff
@@ -16,11 +16,11 @@
     ],
     "require": {
         "php": "^7.2",
-        "symfony/framework-bundle": "^3.4 || ^4.2",
+        "symfony/framework-bundle": "^3.4 || ^4.2 || ^5.0",
         "solarium/solarium": "^5.0"
     },
     "require-dev": {
-        "symfony/phpunit-bridge": "^4.3"
+        "symfony/phpunit-bridge": "^4.3 || ^5.0"
```
Comment (dmaicher) [inline composer.json:23]: here we can actually only require `^5.0`

**Comment by dmaicher** [comment]
I would also suggest to adjust the travis build to test all 3 major Symfony versions explicitly.

This could be done after https://github.com/nelmio/NelmioSolariumBundle/pull/99 is merged

**Comment by dmaicher** [comment]
@thesunrise1983 could you add Symfony 5 to the matrix here?

https://github.com/nelmio/NelmioSolariumBundle/blob/master/.travis.yml#L12

With PHP 7.4 should be enough I think :blush:

File: composer.json (hunk 1 above)
Comment (dmaicher) [reply composer.json:23]: To clarify why: `symfony/phpunit-bridge` is not listed within `symfony/symfony` and does not have any requirements to other packages. So we always just use the latest version here :blush:

**Comment by mhitza** [comment]
Can a maintainer make the last change mentioned and merge this PR? I need to fork the repository in order to use this bundle for now.

File: composer.json (hunk 1 above)
Comment (thesunrise1983) [reply composer.json]: Done. Thanks for the information :-)

**Comment by thesunrise1983** [comment]
> @thesunrise1983 could you add Symfony `5.0.*` to the matrix here?
> 
> https://github.com/nelmio/NelmioSolariumBundle/blob/master/.travis.yml#L12
> 
> With PHP 7.4 should be enough I think blush

done. I did my best ;-)

**Comment by dmaicher** [comment]
@Seldaek can we move forward with this? :blush:

**Comment by dmaicher** [comment]
@thesunrise1983 @mhitza not sure if this will ever get released.

I decided to maintain my own bundle instead: https://packagist.org/packages/dama/solarium-bundle

Its fully compatible with Symfony 5 and requires the latest Solarium 5.2

**Comment by thePanz** [comment]
Sorry for the late responses @thesunrise1983  and @dmaicher .
I will merge this into master, thank you!

To @dmaicher I noticed that you introduced phpstan into your fork, would you like to contribute it here and merge again your fork?

I am willing to add you as co-maintainer of this bundle/repository, if @Seldaek agrees :)

**Comment by dmaicher** [comment]
> Sorry for the late responses @thesunrise1983 and @dmaicher .
> I will merge this into master, thank you!
> 
> To @dmaicher I noticed that you introduced phpstan into your fork, would you like to contribute it here and merge again your fork?
> 
> I am willing to add you as co-maintainer of this bundle/repository, if @Seldaek agrees :)

@thePanz I would be happy to help maintaining this bundle instead :wink: You can also reach me on Symfony Slack if you want to discuss things.

**Pull request merged**